### PR TITLE
Benchmark Reorganization

### LINKED ISSUE #468: Move benchmark/Array* files to benchmark/Streamly/Benchmark/Memory/*

### PR DESCRIPTION
**Pull request by adithyaov**
Closes #468

### CODE REVIEW

**Comment by adithyaov** [comment]
We can potentially move nested unfolds out of lib. I think it's better to add an appropriate bench group regarding the mem restriction as well in the files that have been renamed.

File: benchmark/Streamly/Benchmark/Data/NestedUnfoldOps.hs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 -- License     : BSD3
 -- Maintainer  : streamly@composewell.com
 
-module NestedUnfoldOps where
+module Streamly.Benchmark.Data.NestedUnfoldOps where
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Data/NestedUnfoldOps.hs:8]: 1. Rename this to `Data.Unfold` to reflect the path of the module in the library.
2. Move the unfold benchmarks from Prelude here.

File: benchmark/streamly-benchmarks.cabal [hunk 2]
```diff
@@ -324,7 +324,7 @@ benchmark linear-async
   import: bench-options-threaded
   type: exitcode-stdio-1.0
   ghc-options: -with-rtsopts "-T -N2 -K64K -M16M"
-  hs-source-dirs: .
+  hs-source-dirs: ., Streamly/Benchmark/Prelude
```
Comment (harendra-kumar) [inline benchmark/streamly-benchmarks.cabal:327]: is "." still needed here?

Comment (adithyaov) [reply benchmark/streamly-benchmarks.cabal:327]: You're right, I'll be removing it.

File: benchmark/Streamly/Benchmark/Data/NestedUnfoldOps.hs (hunk 1 above)
Comment (adithyaov) [reply benchmark/Streamly/Benchmark/Data/NestedUnfoldOps.hs:8]: Yes, I've done it already, didn't push it yet.

File: benchmark/Chart.hs [hunk 3]
```diff
@@ -423,3 +427,13 @@ main = do
                             (makeGraphs "fold-o-n-heap")
                             "charts/fold-o-n-heap/results.csv"
                             "charts/fold-o-n-heap"
+                UnfoldO1Space -> benchShow opts cfg
```
Comment (harendra-kumar) [inline benchmark/Chart.hs:430]: This is all mechanically repeated code, at some point we have to do something about it, maybe use TH or some other way to remove duplication.

**Pull request merged**